R/watcher.R

```diff
@@ -17,8 +17,23 @@ watchout <- function(handler = new_output_handler(),
   push <- function(value) {
     output[i] <<- list(value)
     i <<- i + 1
+
+    switch(output_type(value),
```

Comment:
The watcher is now in charge of calling the handler when we push an output onto the stack.